Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+        
+        //
+        // we haven't yet notified user of start or failure
+        notifiedStartOrFailure = false
+        
+        //
+        // set intitial orientationQuat
+        orientationChanged(notification: nil)
+    }
+    
+    // MARK: Public
+    
+    /// Determines the map point for the given screen point
+    ///
+    /// - Parameter screenPoint: the point in screen coordinates
+    /// - Returns: the map point corresponding to screenPoint
+    public func arScreenToLocation(screenPoint: AGSPoint) -> AGSPoint {
+        return AGSPoint(x: 0.0, y: 0.0, spatialReference: nil)
+    }
+    
+    /// Resets the device tracking, using originCamera if it's not nil or the device's GPS location via the locationManager
+    public func resetTracking() {
+        initialLocation = nil
+        startTracking()
+    }
+
+    /// Resets the device tracking, using the device's GPS location via the locationManager
+    ///
+    /// - Returns: reset operation success or failure
+    public func resetUsingLocationServices() -> Bool {
+        return false
+    }
+    
+    /// Resets the device tracking using a spacial anchor
+    ///
+    /// - Returns: reset operation success or failure
+    public func resetUsingSpatialAnchor() -> Bool {
+        return false
+    }
+    
+    /// Starts device tracking
+    public func startTracking() {
+        
+        if !isSupported {
+            didStartOrFailWithError(ArcGISARView.notSupportedError())
+            return
+        }
+        
+        if let origin = originCamera {
+            //
+            // we have a starting camera
+            initialTransformationMatrix = origin.transformationMatrix
+            sceneView.setViewpointCamera(origin)
+            finalizeStart()
+        }
+        else {
+            //
+            // no starting camera, use location manger to get initial location
+            let authStatus = CLLocationManager.authorizationStatus()
+            switch authStatus {
+            case .notDetermined:
+                startWithAccessNotDetermined()
+            case .restricted, .denied:
+                startWithAccessDenied()
+            case .authorizedAlways, .authorizedWhenInUse:
+                startWithAccessAuthorized()
+            }
+        }
+        
+        //
+        // we need to know when the device orientation changes in order to update the Camera transformation
+        UIDevice.current.beginGeneratingDeviceOrientationNotifications()
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.orientationChanged(notification:)),
+            name: UIDevice.orientationDidChangeNotification,
+            object: nil
+        )
+        
+        //
+        // reset frameCount and start timer to capture frame rate
+        frameCount = 0
+        frameCountTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true, block: { [weak self] (timer) in
```

Comment:
I was just about to submit my review when I happened to notice that Apple provided the code to do this without a `Timer` in the Platforms State of the Union:

![IMG_2416](https://user-images.githubusercontent.com/2257493/58837692-2f70d300-8611-11e9-8b48-4e7c1849890e.jpg)

Reply:
I'll give that a try, thanks!